### PR TITLE
Update GLExtensions.cpp

### PR DESCRIPTION
**Pull request by Duron27**
add GLES3

### CODE REVIEW

**Comment by AnyOldName3** [comment]
This PR is pretty huge and definitely breaks things. E.g. it totally swaps out loads of OpenThreads stuff for STL threading primitives, but those aren't even available in C++03, which is what OSG uses. That particular change might be better dealt with by adding STL stuff as yet another backend for OpenThreads, or just figuring out how to make one of the several existing backends work (and at least one of them will do).

**Comment by elsid** [comment]
For openmw we use C++11 to get move constructor for `osg::ref_ptr` but if there is no need to make a change, don't do it. And every change should be described in a separate commit message. What the change does and why it's required (what kind of problem it solves). If you apply a patch, reference it in the commit message with URL that leads to the patch of specific version. Commits with messages like "patches added" and "Update GLExtensions.cpp" don't explain what they do an why and very confusing for anyone looking at them without knowing the context.

Also this is blocked on the ability for CI to run on pull requests: https://github.com/OpenMW/osg/pull/27 .

File: CMakeLists.txt [hunk 1]
```diff
@@ -735,7 +735,7 @@ OPTION(BUILD_OSG_PLUGINS "Build OSG Plugins - Disable for compile testing exampl
 mark_as_advanced(BUILD_OSG_PLUGINS)
 ################################################################################
 # 3rd Party Dependency Stuff
-OPTION(OSG_FIND_3RD_PARTY_DEPS "Enable to search for Android or Windows dependencies in ./3rdparty" ON)
+OPTION(OSG_FIND_3RD_PARTY_DEPS "Enable to search for Android or Windows dependencies in ./3rdparty" OFF)
```
Comment (elsid) [inline CMakeLists.txt:738]: This likely will break all builds not setting this option explicitly. Set it in your cmake command not here.

File: CMakeModules/FindCOLLADA.cmake [hunk 2]
```diff
@@ -19,7 +19,7 @@ SET( COLLADA_ENV_VAR_AVAILABLE $ENV{COLLADA_DIR} )
 IF ( COLLADA_ENV_VAR_AVAILABLE )
     SET(COLLADA_DOM_ROOT "$ENV{COLLADA_DIR}/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
 ELSE ()
-    SET(COLLADA_DOM_ROOT "${ACTUAL_3DPARTY_DIR}/include/1.4/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
+    SET(COLLADA_DOM_ROOT "${ACTUAL_3DPARTY_DIR}/include/collada-dom2.5/1.4/dom" CACHE PATH "Location of Collada DOM directory" FORCE)
```
Comment (elsid) [inline CMakeModules/FindCOLLADA.cmake:22]: Collada 1.4 is unlikely located in the 2.5 unless something weird is done to setup packages. Please don't make this change.

File: CMakeModules/FindCOLLADA.cmake [hunk 3]
```diff
@@ -74,6 +74,7 @@ FIND_PATH(COLLADA_INCLUDE_DIR dae.h
     /opt/include
     /usr/freeware/include
     ${ACTUAL_3DPARTY_DIR}/include
+    ${ACTUAL_3DPARTY_DIR}/include/collada-dom2.5/
```
Comment (elsid) [inline CMakeModules/FindCOLLADA.cmake:77]: There is a path suffix `collada-dom2.5` just below which should make it possible to find collada 2.5. No need to have this change.

File: CMakeModules/OsgAndroidMacroUtils.cmake [hunk 4]
```diff
@@ -163,7 +163,7 @@ MACRO(ANDROID_3RD_PARTY)
     )
     #set(ENV{AND_OSG_LIB_NAMES} "$ENV{AND_OSG_LIB_NAMES} libft2")
     #set(ENV{AND_OSG_LIB_PATHS} "$ENV{AND_OSG_LIB_PATHS}include ${FREETYPE_DIR}/Android.mk \n")
-    set(FREETYPE_INCLUDE_DIRS ${FREETYPE_DIR}/include ${FREETYPE_DIR}/include/freetype/config)
+    set(FREETYPE_INCLUDE_DIRS ${FREETYPE_DIR}/include ${FREETYPE_DIR}/include/freetype2)
```
Comment (elsid) [inline CMakeModules/OsgAndroidMacroUtils.cmake:166]: This likely will break builds with `freetype`. What kind of problem this solves?

File: CMakeModules/OsgMacroUtils.cmake [hunk 5]
```diff
@@ -301,6 +301,10 @@ MACRO(SETUP_PLUGIN PLUGIN_NAME)
         ADD_LIBRARY(${TARGET_TARGETNAME} STATIC ${TARGET_SRC} ${TARGET_H})
     ENDIF(DYNAMIC_OPENSCENEGRAPH)
 
+    # XXX: remove the "lib" prefix e.g. "libosgdb_bmp.a" => "osgdb_bmp.a"
+    # for some reason this is only a problem on android
+    SET_TARGET_PROPERTIES(${TARGET_TARGETNAME} PROPERTIES PREFIX "")
```
Comment (elsid) [inline CMakeModules/OsgMacroUtils.cmake:306]: What is the problem?

File: src/osgPlugins/CMakeLists.txt [hunk 6]
```diff
@@ -161,9 +161,13 @@ IF(NOT OSG_GLES1_AVAILABLE AND NOT OSG_GLES2_AVAILABLE)
     ENDIF()
 ENDIF()
 
-IF(COLLADA_FOUND AND OSG_CPP_EXCEPTIONS_AVAILABLE)
+#fix me
+#IF(COLLADA_FOUND AND OSG_CPP_EXCEPTIONS_AVAILABLE)
+#    ADD_PLUGIN_DIRECTORY(dae)
+#ENDIF()
```
Comment (elsid) [inline src/osgPlugins/CMakeLists.txt:167]: What is this for?

**Review by elsid — CHANGES_REQUESTED** [review]
So far linking openmw to this version of osg leads to a difference in the rendered picture:

This MR:

screenshot385

Upstream version:

screenshot386

**Comment by Duron27** [comment]
Woa sorry,  I had forgotten about this MR and added that patch for testing.  I did not mean to include those in the MR at all

File: CMakeModules/FindCOLLADA.cmake (hunk 3 above)
Comment (AnyOldName3) [reply CMakeModules/FindCOLLADA.cmake:77]: I'm pretty sure we're married to Collada DOM 2.4. 2.5 is really an alpha of the de-facto abandoned 3.x series, so has breaking changes.

File: CMakeLists.txt (hunk 1 above)
Comment (Duron27) [reply CMakeLists.txt:738]: I did not mean to add that commit from a few days ago.  Can it be rolled back?

**Pull request closed without merging**